cache-filter/src/configuration/filter.rs

```diff
@@ -0,0 +1,31 @@
+use serde::Deserialize;
+use std::time::Duration;
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(default)]
+pub struct FilterConfig {
+    /// Threescale cluster name that indicates threescale backend that includes SM API. Should provide the cluster
+    /// name of the threescale cluster in the envoy.yaml file.
+    threescale_cluster: String,
+
+    /// The basepath for the authorize endpoint.
+    threescale_auth_basepath: String,
```

Comment:
@unleashed yeah since 3scale has a GET endpoint with queries, we can't just point a cluster , an endpoint and send a request. What is this "3scale ids" filter ?? I thought we have [AUTH] -> [Cache] -> [Ratelimit]. I had a look at the auth filter and it builds up the message with uri to send.

Reply:
> @unleashed yeah since 3scale has a GET endpoint with queries, we can't just point a cluster , an endpoint and send a request.

What I mean is that you have a client that will take the base URL and the 3scale ids you want, and whether you want authorize, authrep or report, and it'll provide you with everything you need to pass in to Envoy's own HTTP calling system, without you needing to do any coding around paths and parameters.

> What is this "3scale ids" filter ?? I thought we have [AUTH] -> [Cache] -> [Ratelimit]. I had a look at the auth filter and it builds up the message with uri to send.

It's what we've called the "auth" one, except it must be modified to support (when configured to do so) just outputting to metadata what will become your inputs (instead of calling into 3scale).

Reply:
fixed